File: 30 Days of Code/PHP/Day 28 - RegEx, Patterns, and Intro to Databases.php
<?php



$N = intval(trim(fgets(STDIN)));
$name = array();
for ($N_itr = 0; $N_itr < $N; $N_itr++) {
    $first_multiple_input = explode(' ', rtrim(fgets(STDIN)));

    $firstName = $first_multiple_input[0];

    $emailID = $first_multiple_input[1];
    
    if (preg_match("/@gmail.com/", $emailID)){
        array_push($name,$firstName);}
            
    
}

sort($name);

$itr = 0;
while($itr<count($name)){
    print_r($name[$itr]."\n");
    $itr++;
}

?>

File: 30 Days of Code/Python/Day 28 - RegEx, Patterns, and Intro to Databases.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    N = int(input().strip())
    name = []
    for N_itr in range(N):
        first_multiple_input = input().rstrip().split()

        firstName = first_multiple_input[0]

        emailID = first_multiple_input[1]
        
        if re.search(".+@gmail\.com$",emailID):
            name.append(firstName)
            
    name.sort()
    print(*name,sep="\n")